parley/src/context.rs

```diff
@@ -102,8 +101,32 @@ impl<B: Brush> LayoutContext<B> {
     pub(crate) fn analyze_text(&mut self, text: &str) {
         let text = if text.is_empty() { " " } else { text };
         let mut a = swash::text::analyze(text.chars());
-        for x in a.by_ref() {
-            self.info.push((CharInfo::new(x.0, x.1), 0));
+
+        let mut word_break = Default::default();
+        let mut style_idx = 0;
+
+        let mut char_indices = text.char_indices();
+        loop {
+            let Some((char_idx, _)) = char_indices.next() else {
+                break;
+            };
+
+            // Find the style for this character. If the text is empty, we may not have any styles. Otherwise,
+            // self.styles should span the entire range of the text.
+            while let Some(style) = self.styles.get(style_idx) {
+                if style.range.end > char_idx {
+                    word_break = style.style.word_break;
+                    break;
+                }
+                style_idx += 1;
+            }
+            a.set_break_strength(word_break);
+
+            let Some((properties, boundary)) = a.next() else {
+                break;
+            };
+
+            self.info.push((CharInfo::new(properties, boundary), 0));
         }
```

Comment:
This seems pretty awkward, but I guess fine...

Reply:
Yeah, it's a bit unfortunate. I thought about other ways to do it in Swash, e.g. passing in the word break strength with each call to `next`, but that would be a breaking change there and make it no longer an iterator.

Reply:
I think you could make the analysis take an iterator over `(char, WordBreakStrength)` rather than just `char`? And you could have a different constructor for backwards compat.